specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/applicationGateway.json

```diff
@@ -1007,10 +1008,12 @@
           "description": "Name of the IP configuration that is unique within an Application Gateway."
         },
         "etag": {
+          "readOnly": true,
           "type": "string",
           "description": "A unique read-only string that changes whenever the resource is updated."
         },
         "type": {
+          "readOnly": true,
```

Comment:
just curious, type is readonly?

Reply:
Yes, they are set by the server (e.g. `Microsoft.Network/applicationGateways`)